sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -357,17 +382,23 @@ export class EventHubSender extends LinkEntity {
         await defaultLock.acquire(this.senderLock, () => {
           return this._init();
         });
+        if (abortSignal) {
+          abortSignal.removeEventListener("abort", onAbort);
+        }
+        resolve(this._sender!.maxMessageSize);
+      } catch (err) {
+        log.error(
+          "[%s] An error occurred while creating the sender %s",
+          this._context.connectionId,
+          this.name,
+          err
+        );
+        if (abortSignal) {
+          abortSignal.removeEventListener("abort", onAbort);
+        }
```

Comment:
Instead of repeating this in both success and failure, you can use the `finally` block